app/controllers/users_controller.rb

```diff
@@ -19,7 +19,7 @@ def edit
   def update
     set_tabs(params["user"]["tab"])
     if @user.update(permitted_attributes(@user))
-      RssReaderFetchUserJob.perform_later(@user.id)
+      RssReaderFetchUserWorker.perform_async(@user.id)
```

Comment:
While we are here to make this more performant, how about we check if a feed_url is present here before we even enqueue the worker? That way we skip all the Redis and user lookup calls if there is no feed_url
```ruby
RssReaderFetchUserWorker.perform_async(@user.id) if @user.feed_url.present?
```

Reply:
Do we even need to fetch it if the `feed_url` exists but was not updated? I haven't checked how exactly fetching from rss works in this case in details.

Reply:
Not sure, I'm not an RSS person, probably safe to enqueue the job if we have the `feed_url` and then possibly further optimize this in the future. @nickytonline or @rhymes would either of you possibly know?

Reply:
The masters of the RSS feed logic here are @maestromac and @Zhao-Andy 
I honestly don't know :D

Reply:
I went ahead and put in the check as @mstruve suggested with the intention of re-visiting in the future to optimize it!

Reply:
That guard is definitely useful here. We kept this the way it is for the following reason on `/settings/publish-from-rss`
> Your feed will be fetched every time you submit this form and updates will be automatically fetched periodically thereafter. Contact yo@dev.to if you encounter issues.

I actually thought it was limited to only that page.